main.cpp

```diff
@@ -28,510 +28,558 @@ Thing: Car Wash
 {
+    if(numberOfHeldNotes)
+    {
+        return numberOfHeldNotes;
+    }
+        return playNothing(0);
+}
 
-//    1) key depressed
+int Chord::playNothing(int numOfNotes = 0)
+{
+    return numOfNotes;
+}
+
+struct Keys
+{
     bool keyDepressed = false;
-//    2) velocity of key pressed (1-127)
     int noteVelocity = 67;
-//    3) MIDI number of note pressed (1-127)
     int noteMidiNumber = 59;
-//    4) note legato value (note held, while another is played)
-    float legatoValue = 1003.5;
-//    5) keyboard set to monophonic
+    float glideValue = 1003.5;
     bool monophonicSynth = true;
 
-//    1) play legato
-    void playLegato();
-//    2) play chords
-    void playChords();
-//    3) play monophonic
-    void playMonophonic(Keys keys);
+    bool playGlide(float glideValue);
+    void playChords(Keys keys);
+    bool playMonophonic(bool monophonicSynth);
 };
 
-/*
-Thing 6) Oscillator
-5 properties:
-    1) pitch value of oscillator
-    2) octave setting of oscillator (int 'foot' selection of LFO, 32, 16, 8, 4, 2)
-    3) oscillator volume level
-    4) oscillator pitch source
-    5) waveform of oscillator
-3 things it can do:
-    1) make a sound
-    2) be an LFO
-    3) be a square wave
- */
+bool Keys::playGlide(float glideNumber)
+{
+    return playMonophonic(glideNumber > 1.0f);    
+}
+
+void Keys::playChords(Keys keys)
+{
+    keys.playMonophonic(false);
+}
+
+bool Keys::playMonophonic(bool monophonicKeys = true)
+{
+    if (monophonicKeys)
+    {
+        return monophonicSynth;
+    }
+    return monophonicKeys;    
+}
 
 struct Oscillator
 {
-//    1) pitch value of oscillator (Hz)
     int oscillatorPitch = 875;
-//    2) octave setting of oscillator (int 'foot' selection of LFO, 32, 16, 8, 4, 2)
     unsigned int oscillatorOctave = 16;
-//    3) oscillator volume level (dB)
     int oscillatorLevel = -6;
-//    4) oscillator pitch source
     std::string oscillatorPitchSource = "Keyboard";
-//    5) waveform of oscillator
     std::string oscillatorWaveform = "pulse";
 
-//    1) make a sound
-    void makeSound(Keys keys);
-//    2) be an LFO
-    void makeOscillatorAnLFO(int oscillatorNumber);
-//    3) be a square wave
-    void makeOscillatorSquareWave(int oscillatorNumber);
-
+    int makeSound(int oscillatorPitch);
+    void makeOscillatorAnLFO(int oscillatorFrequency);
+    int makeOscillatorSquareWave(int oscillatorWave);
 };
 
-/*
-Thing 7) Filter
-5 properties:
-    1) Cutoff frequency
-    2) Resonance
-    3) Attack value
-    4) Decay value
-    5) Release value
-3 things it can do:
-    1) change oscillator cutoff
-    2) change oscillator resonance
-    3) receive host automation
- */
+int Oscillator::makeSound(int oscillatorFrequency)
+{
+    return oscillatorFrequency;
+}
+
+void Oscillator::makeOscillatorAnLFO(int oscillatorFrequency)
+{
+    oscillatorFrequency = 15;
+}
+
+int Oscillator::makeOscillatorSquareWave(int oscillatorWave)
+{
+    return oscillatorWave = 2;
+}
```

Comment:
What value is this function returning?
make it obvious.
Break the habit of writing `return something modified by something else`
instead, write: `something modified by something else` then `return something` on a separate line:
```
oscillatorWave = 2;
return oscillatorWave;
```
- [ ] revise

Reply:
Done.